akka-persistence-query/src/test/scala/akka/persistence/query/PersistenceQuerySpec.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.persistence.query
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.ActorSystem
+import akka.persistence.journal.{ EventAdapter, EventSeq }
+import com.typesafe.config.ConfigFactory
+import org.scalatest.{ BeforeAndAfterAll, Matchers, WordSpecLike }
+import org.scalautils.ConversionCheckedTripleEquals
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class PersistenceQuerySpec extends WordSpecLike with Matchers with BeforeAndAfterAll with ConversionCheckedTripleEquals {
+
+  val anything: Query[String, _] = null
+
+  val eventAdaptersConfig =
+    s"""
+      |akka.persistence.query.journal.mock {
+      |  event-adapters {
+      |    adapt = ${classOf[PrefixStringWithPAdapter].getCanonicalName}
+      |  }
+      |}
+    """.stripMargin
+
+  "ReadJournal" must {
+    "be found by full config key" in {
+      withActorSystem() { system ⇒
+        PersistenceQuery.get(system).readJournalFor(MockReadJournal.Identifier)
+      }
+    }
+
+    "throw if unable to find query journal by config key" in {
+      withActorSystem() { system ⇒
+        intercept[IllegalArgumentException] {
+          PersistenceQuery.get(system).readJournalFor(MockReadJournal.Identifier + "-unknown")
+        }.getMessage should include("missing persistence read journal")
+      }
+    }
```

Comment:
we’ll need to include a somewhat real implementation so that we can also write some sensible tests for the whole infrastructure